airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -154,13 +154,19 @@ def get_json_schema(self) -> Mapping[str, Any]:
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
         r = response.json()
 
-        if all(key in r for key in ["limit", "offset", "rowCount"]):
-            limit, offset, total_rows = r["limit"], r["offset"], r["rowCount"]
+        if "rowCount" in r:
+            total_rows = r["rowCount"]
 
-            if total_rows <= offset:
-                return None
+            if self.page_size is None:
+                self.page_size = 100000
```

Comment:
This is confusing to me. Why we're increasing the page size value? This should be a const value right?
Can you create a constant for the `10000` called PAGE_SIZE = 10000 and add the documentation link to this value? https://developers.google.com/analytics/devguides/reporting/core/v3/reference#maxResults this is the correct one? Should it 10k

IMHO you must rename the variable to `offset` instead of `self.page_size`.

Reply:
I agree with self.page_size replacing to offset. Resolved it next commit